From cf79bfccf2e1fa71a602c17b96e1df44e916b9ac Mon Sep 17 00:00:00 2001
From: John Zhuge <jzhuge@cloudera.com>
Date: Sun, 13 Aug 2017 00:52:03 -0700
Subject: [PATCH 2587/2748] HADOOP-14349. Rename ADLS CONTRACT_ENABLE_KEY.
 Contributed by Mingliang Liu

(cherry picked from commit e4c32bccde3147f51263cb58cdb480453b1cb3e9)

Change-Id: I1817d9797fc208226a7f354cd0a368496b227af8
---
 .../src/site/markdown/index.md                     |    4 ++--
 .../fs/adl/live/AdlStorageConfiguration.java       |    8 +++++++-
 2 files changed, 9 insertions(+), 3 deletions(-)

diff --git a/hadoop-tools/hadoop-azure-datalake/src/site/markdown/index.md b/hadoop-tools/hadoop-azure-datalake/src/site/markdown/index.md
index d2da858..d4b7d8e 100644
--- a/hadoop-tools/hadoop-azure-datalake/src/site/markdown/index.md
+++ b/hadoop-tools/hadoop-azure-datalake/src/site/markdown/index.md
@@ -253,7 +253,7 @@ Add the following properties to `core-site.xml`
 ```
 ## Testing the azure-datalake-store Module
 The `hadoop-azure` module includes a full suite of unit tests.
-Most of the tests will run without additional configuration by running mvn test.
+Most of the tests will run without additional configuration by running `mvn test`.
 This includes tests against mocked storage, which is an in-memory emulation of Azure Data Lake Storage.
 
 A selection of tests can run against the Azure Data Lake Storage. To run these
@@ -262,7 +262,7 @@ information mentioned in the above sections and the following properties.
 
 ```xml
 <property>
-    <name>dfs.adl.test.contract.enable</name>
+    <name>fs.adl.test.contract.enable</name>
     <value>true</value>
 </property>
 
diff --git a/hadoop-tools/hadoop-azure-datalake/src/test/java/org/apache/hadoop/fs/adl/live/AdlStorageConfiguration.java b/hadoop-tools/hadoop-azure-datalake/src/test/java/org/apache/hadoop/fs/adl/live/AdlStorageConfiguration.java
index 7d6153d..a9b4ab8 100644
--- a/hadoop-tools/hadoop-azure-datalake/src/test/java/org/apache/hadoop/fs/adl/live/AdlStorageConfiguration.java
+++ b/hadoop-tools/hadoop-azure-datalake/src/test/java/org/apache/hadoop/fs/adl/live/AdlStorageConfiguration.java
@@ -35,7 +35,7 @@
   static final String CONTRACT_XML = "adls.xml";
 
   private static final String CONTRACT_ENABLE_KEY =
-      "dfs.adl.test.contract.enable";
+      "fs.adl.test.contract.enable";
   private static final boolean CONTRACT_ENABLE_DEFAULT = false;
 
   private static final String FILE_SYSTEM_KEY =
@@ -49,6 +49,12 @@
   private static boolean isContractTestEnabled = false;
   private static Configuration conf = null;
 
+  static {
+    Configuration.addDeprecation("dfs.adl.test.contract.enable",
+        CONTRACT_ENABLE_KEY);
+    Configuration.reloadExistingConfigurations();
+  }
+
   private AdlStorageConfiguration() {
   }
 
-- 
1.7.9.5

